testApi/src/test/scala/africa/absa/testing/scapi/rest/response/ResponseExtractTest.scala

```diff
@@ -47,124 +66,93 @@ class ResponseExtractTest extends FunSuite {
+    val assertionParam2: ResponseAction = ResponseAction(group = ResponseActionGroupType.EXTRACT_JSON, name = ExtractJsonResponseActionType.STRING_FROM_LIST, Map("cacheKey" -> "1", "listIndex" -> "", "jsonKey" -> "", "cacheLevel" -> ""))
+    val assertionParam3: ResponseAction = ResponseAction(group = ResponseActionGroupType.EXTRACT_JSON, name = ExtractJsonResponseActionType.STRING_FROM_LIST, Map("cacheKey" -> "1", "listIndex" -> "x", "jsonKey" -> "", "cacheLevel" -> ""))
+    val assertionParam4: ResponseAction = ResponseAction(group = ResponseActionGroupType.EXTRACT_JSON, name = ExtractJsonResponseActionType.STRING_FROM_LIST, Map("cacheKey" -> "1", "listIndex" -> "x", "jsonKey" -> "y", "cacheLevel" -> ""))
 
-    val result = ExtractJsonResponseAction.stringFromList(responseNoJsonBody, cacheKey, listIndex, jsonKey, runtimeCacheLevel)
-    assert(!result)
+    interceptMessage[ContentValidationFailedException]("Content validation failed for value: '': Received string value of 'ExtractJson.string-from-list.cacheKey' is empty.") {
+      ExtractJsonResponseAction.validateContent(assertionParam1)
+    }
+    interceptMessage[ContentValidationFailedException]("Content validation failed for value: '': Received string value of 'ExtractJson.string-from-list.listIndex' is empty.") {
+      ExtractJsonResponseAction.validateContent(assertionParam2)
+    }
+    interceptMessage[ContentValidationFailedException]("Content validation failed for value: '': Received string value of 'ExtractJson.string-from-list.jsonKey' is empty.") {
+      ExtractJsonResponseAction.validateContent(assertionParam3)
+    }
+    interceptMessage[ContentValidationFailedException]("Content validation failed for value: '': Received string value of 'ExtractJson.string-from-list.cacheLevel' is empty.") {
+      ExtractJsonResponseAction.validateContent(assertionParam4)
+    }
   }
 
-  test("stringFromList - incorrect parameters - no json arrays in response body") {
-    val cacheKey = "question_id"
-    val listIndex = 0
-    val jsonKey = "id"
-    val runtimeCacheLevel = "Test"
-
-    val result = ExtractJsonResponseAction.stringFromList(responseJsonNoArrayBody, cacheKey, listIndex, jsonKey, runtimeCacheLevel)
-    assert(!result)
+  test("validateContent - unsupported option") {
+    interceptMessage[UndefinedResponseActionTypeException]("Undefined response action content type: 'Unsupported'") {
+      ExtractJsonResponseAction.validateContent(assertionUnsupported)
+    }
   }
 
   /*
-    validateStringFromList
+    performResponseAction
    */
 
-  // positive test "stringFromList - correct parameters" - tested during "validateContent - STRING_FROM_LIST"
+  test("performAssertion - STRING_FROM_LIST") {
+    val result: Try[Unit] = ExtractJsonResponseAction.performResponseAction(responseWithID, assertionStringFromList)
+    assert(result.isSuccess)
+    assertEquals("382be85a-1f00-4c15-b607-cbda03ccxxx2", RuntimeCache.get("question_id").get)
```

Comment:
dtto, swapped values (actual, expected)?

Reply:
Wrong order is a bug. Fixed in commit **TBD**.

Reply:
After deeper analysis I have decided to follow the framework order to avoid wrong outputs.